atlassian_jwt_auth/frameworks/common/backend.py

```diff
@@ -0,0 +1,95 @@
+from abc import ABCMeta, abstractmethod, abstractproperty
+
+from atlassian_jwt_auth import HTTPSPublicKeyRetriever, JWTAuthVerifier
+
+from jwt.compat import text_type
+from .utils import SettingsDict
+
+
+class Backend():
+    """Abstract class representing a web framework backend
+
+    Backends allow specific implementation details of web frameworks to be
+    abstracted away from the underlying logic of ASAP.
+    """
+    __metaclass__ = ABCMeta
+
+    default_headers_401 = {'WWW-Authenticate': 'Bearer'}
+    default_settings = {
+        # The class to be instantiated to retrieve public keys
+        'ASAP_KEY_RETRIEVER_CLASS': HTTPSPublicKeyRetriever,
+
+        # The repository URL where the key retriever can fetch public keys
+        'ASAP_PUBLICKEY_REPOSITORY': None,
+
+        # Whether or not ASAP authentication is required
+        # This is primarily useful when phasing in ASAP authentication
+        'ASAP_REQUIRED': True,
+
+        # The valid audience valuw expected when authenticating tokens
+        'ASAP_VALID_AUDIENCE': None,
+
+        # The amount of leeway to apply when evaluating token expiration
+        # timestamps
+        'ASAP_VALID_LEEWAY': 0,
+
+        # A list of valid token issuers allowed to authenticate
```

Comment:
Ideally this would be a set.

Reply:
Agreed. I'll change it to say an `iterable` and then create a set inside when constructing the final settings dict.